### ISSUE TITLE
Source Code Disclosure (WEB-INF folder) fails with Java 9+

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#4038]
The scanner "Source Code Disclosure - /WEB-INF folder" fails to run with Java 9:
```
113913 [ZAP-ActiveScanner-0] ERROR org.zaproxy.zap.ZAP$UncaughtExceptionLogger  - Exception in thread "ZAP-ActiveScanner-0"
java.lang.NoClassDefFoundError: sun/misc/URLClassPath
	at com.strobel.assembler.metadata.ClasspathTypeLoader.<init>(ClasspathTypeLoader.java:66)
	at com.strobel.assembler.metadata.ClasspathTypeLoader.<init>(ClasspathTypeLoader.java:42)
	at com.strobel.assembler.InputTypeLoader.<init>(InputTypeLoader.java:45)
	at com.strobel.decompiler.Decompiler.decompile(Decompiler.java:39)
	at org.zaproxy.zap.extension.ascanrulesBeta.SourceCodeDisclosureWEBINF.scan(SourceCodeDisclosureWEBINF.java:212)
	at org.parosproxy.paros.core.scanner.AbstractPlugin.run(AbstractPlugin.java:379)
	at java.base/java.lang.Thread.run(Thread.java:844)
Caused by: java.lang.ClassNotFoundException
	at org.zaproxy.zap.control.AddOnClassLoader.findClass(AddOnClassLoader.java:256)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:563)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:496)
	... 7 more
```

This is a known issue with the library:
~~https://bitbucket.org/mstrobel/procyon/issues/320/java-9-sunmiscurlclasspath-and~~ mstrobel/procyon#12

Add-on:
Active scanner rules (beta), version 21.

Related to #2602.

**Comment by kingthorin** [comment]
Note the project has moved from BitBucket to GitHub:
https://github.com/mstrobel/procyon/issues/12

**Comment by thc202** [comment]
Fixed in zaproxy/zap-extensions#2338.
